packages/nimble-components/src/select/index.ts

```diff
@@ -216,6 +216,9 @@ export class Select
     private _value = '';
     private forcedPosition = false;
     private openActiveIndex?: number;
+    private readonly selectedOptionObserver? = new MutationObserver(() => {
```

Comment:
I might be missing something but why does this type need to handle the mutation observer possibly being `undefined`?

Reply:
It's because the `selectedIndexChanged` callback will run prior to the constructor or even the assignment of this field, and at that time `selectedOptionObserver` is `undefined`. While I admit this looks a bit goofy, it's at least declaratively honest. The only way I could see to make this less "goofy" looking is to just declare it here, and then define it in `connectedCallback`. I don't have a strong opinion.

Reply:
Makes sense, I'm fine to leave it as-is now that I understand the rationale.